otter/test/util/test_zkpartitioner.py

```diff
@@ -122,12 +122,44 @@ def test_repeat(self):
         self.partitioner.startService()
         self.log.msg.assert_called_once_with(
             'Got buckets {buckets}',
-            buckets=[2, 3], path=self.path,
+            buckets=[2, 3], old_buckets=[], path=self.path,
             otter_msg_type='partition-acquired')
         self.clock.advance(10)
         self.clock.advance(10)
         self.assertEqual(self.buckets_received, [[2, 3], [2, 3], [2, 3]])
 
+    def test_no_log_spam(self):
+        """
+        Bucket changes are only logged when the buckets actually change.
+        """
+        self.kz_partitioner.acquired = True
+        self.kz_partitioner.__iter__.return_value = [2, 3]
+        self.partitioner.startService()
+        self.clock.advance(10)
+        self.clock.advance(10)
+        self.log.msg.assert_called_once_with(
+            'Got buckets {buckets}',
+            buckets=[2, 3], old_buckets=[], path=self.path,
+            otter_msg_type='partition-acquired')
+
+    def test_no_log_spam(self):
```

Comment:
These two test names are the same, which means that one will will not run.  Might be useful to differentiate the docstrings too?  One is testing that if there is no change, there is only the initial bucket allocation log, and the other testing that when there are changes after the initial bucket allocation, it is logged.